client/mm/mm_simple_arb.go

```diff
@@ -0,0 +1,547 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package mm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"sync/atomic"
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/mm/libxc"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/calc"
+	"decred.org/dcrdex/dex/order"
+)
+
+// SimpleArbConfig is the configuration for an arbitrage bot that only places
+// orders when there is a profitable arbitrage opportunity.
+type SimpleArbConfig struct {
+	// CEXName is the name of the cex that the bot will arbitrage.
+	CEXName string `json:"cexName"`
+	// ProfitTrigger is the minimum profit before a cross-exchange trade
+	// sequence is initiated. Range: 0 < ProfitTrigger << 1. For example, if
+	// the ProfitTrigger is 0.01 and a trade sequence would produce a 1% profit
+	// or better, a trade sequence will be initiated.
+	ProfitTrigger float64 `json:"profitTrigger"`
+	// MaxActiveArbs sets a limit on the number of active arbitrage sequences
+	// that can be open simultaneously.
+	MaxActiveArbs uint32 `json:"maxActiveArbs"`
+	// NumEpochsLeaveOpen is the number of epochs an arbitrage sequence will
+	// stay open if one or both of the orders were not filled.
+	NumEpochsLeaveOpen uint32 `json:"numEpochsLeaveOpen"`
+	// BaseOptions are the multi-order options for the base asset wallet.
+	BaseOptions map[string]string `json:"baseOptions"`
+	// QuoteOptions are the multi-order options for the quote asset wallet.
+	QuoteOptions map[string]string `json:"quoteOptions"`
+}
+
+func (c *SimpleArbConfig) Validate() error {
+	if c.ProfitTrigger <= 0 || c.ProfitTrigger > 1 {
+		return fmt.Errorf("profit trigger must be 0 < t <= 1, but got %v", c.ProfitTrigger)
+	}
+
+	if c.MaxActiveArbs == 0 {
+		return fmt.Errorf("must allow at least 1 active arb")
+	}
+
+	if c.NumEpochsLeaveOpen < 2 {
+		return fmt.Errorf("arbs must be left open for at least 2 epochs")
+	}
+
+	return nil
+}
+
+// arbSequence represents an attempted arbitrage sequence.
+type arbSequence struct {
+	dexOrder       *core.Order
+	cexOrderID     string
+	dexRate        uint64
+	cexRate        uint64
+	cexOrderFilled bool
+	dexOrderFilled bool
+	sellOnDEX      bool
+	startEpoch     uint64
+}
+
+type simpleArbMarketMaker struct {
+	ctx   context.Context
+	host  string
+	base  uint32
+	quote uint32
```

Comment:
I hate to keep asking, but `baseID` and `quoteID` please

Reply:
No problem.